### ISSUE TITLE
Cant login with LDAP

### ISSUE THREAD

**Issue by claudio-azevedo** [ansible-semaphore/semaphore#684]
Hi Guys. I really liked the tool, but I was unable to configure it for login with my Active Directoy account (Microsoft AD). I'm configuring something wrong?

The error:

> INFO[0003] User Test User with email testuser@company.net authorized via LDAP correctly
> 2021/07/27 15:05:14 http: panic serving 10.3.234.48:59256: sql: no rows in result set

My LDAP configuration:

> "ldap_enable": true,
> "ldap_binddn": "ldap@company.net",
>  "ldap_bindpassword": "password",
>  "ldap_server": "ad.company.net:389",
>  "ldap_searchdn": "DC=company,DC=net",
>  "ldap_searchfilter": "(&(sAMAccountName=%s)(memberOf=CN=_GG_ANSIBLE_ADMIN,OU=GLOBAL_GROUPS,OU=GROUPS,DC=company,DC=net))",
>   "ldap_mappings": {
>             "dn": "distinguishedName",
>             "mail": "userPrincipalName",
>             "uid": "sAMAccountName",
>             "cn": "cn"
>         },

In the database (mariadb) I see that only the adm local user exists.
Semaphore should create the user in DB after the authorized log message?

**Comment by DLMSweet** [comment]
I ran into the same problem. It looks like the issue is here:
https://github.com/ansible-semaphore/semaphore/blob/19463ddb4f8ed0420823e62eaef244eed277399d/api/login.go#L131
in my testing that statement never evaluated to true:
`Error is wrong:  sql: no rows in result set != no rows in result set
`
It looks like the errors don't match up for some reason. 
I don't know enough about Go to fix it  though, or if that's really the issue here.

**Comment by graysonwalker99** [comment]
In a previous closed issue from a few years ago, I saw someone had successfully set up ldap with the config below.  I'm still trying to get mine to work, but maybe this will help you if it's a config related issue.

"ldap_enable": true,
"ldap_binddn": "uid=binduser,cn=users,cn=accounts,dc=mydomain,dc=it",
"ldap_bindpassword": "supersecurepassword",
"ldap_server": "server.mydomain.it:389",
"ldap_needtls": false,
"ldap_searchdn": "cn=users,cn=accounts,dc=mydomain,dc=it",
"ldap_searchfilter": "(&(uid=%s)(memberOf=cn=semaphore_users,cn=groups,cn=accounts,dc=mydomain,dc=it))",
 "ldap_mappings": {
 	"dn": "dn",
 	"mail": "mail",
 	"uid": "uid",
 	"cn": "cn"
}

**Comment by DLMSweet** [comment]
It is not a config issue. If I change the code so that statement evaluates to True, it creates the LDAP user and logs in successfully.

**Comment by fiftin** [comment]
@DLMSweet could you please create PR?

**Comment by fiftin** [comment]
@DLMSweet I fould issue source, thank you very mach!

**Comment by fiftin** [comment]
@claudio-azevedo, @DLMSweet sorry for delay, I testing PRs, after that I will create new release.

**Comment by fiftin** [comment]
Hi @claudio-azevedo, @graysonwalker99, @DLMSweet  please try new release https://github.com/ansible-semaphore/semaphore/releases/tag/v2.7.1

**Comment by claudio-azevedo** [comment]
@fiftin now it works. Thanks a lot! I have tested the membership of the group sucessfully.

A little bug that I found now that on the first login (new user), the log shows .... "authorized via LDAP correctly", the user is created on the database, but semaphore thows a page of 401 to this new user, with the following message:

"Error"
"Request failed with status code 401"

There is a button to call the login page again, if the user login again, he is redirected correctly to the "new project page"
